### PR TITLE
split ci into stages

### PR DESCRIPTION
**Pull request by zaro0508**
* Split the travis build into stages: validate, test, and package.
* Use pandoc to convert markdown to rst

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
 cache: pip
 fast_finish: true
+addons:
+  apt_packages:
+    - pandoc
 env:
   global:
     - buildnum=$TRAVIS_BUILD_NUMBER
 install:
   - pip install pre-commit
```
Comment (cascadianblue) [inline .travis.yml:13]: I think this is unnecessary now that "pre-commit" has been added to requirements.txt.

Comment (zaro0508) [reply .travis.yml]: removed.

**Comment by cascadianblue** [comment]
Sorry, I'm confused by the installation of pandoc. Where is it being used, and how is the project already dependent on it?

**Comment by zaro0508** [comment]
@cascadianblue it's being called in the [package target](https://github.com/Sage-Bionetworks/jccli/blob/master/Makefile#L37) in the makefile.  package target calls docs target which uses pandoc.

**Comment by cascadianblue** [comment]
@zaro0508 I see. Okay, everything looks good to me then.

**Pull request merged**